File: api/src/test/java/net/blueshell/api/factory/dto/committee/SimpleCommitteeDTOFactory.java
package net.blueshell.api.factory.dto.committee;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.committee.SimpleCommitteeDTO;
import net.blueshell.api.factory.dto.BaseDtoFactory;
import org.springframework.stereotype.Component;

/**
 * Factory for SimpleCommitteeDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class SimpleCommitteeDTOFactory extends BaseDtoFactory<SimpleCommitteeDTO> {

    @Override
    public Class<SimpleCommitteeDTO> targetType() {
        return SimpleCommitteeDTO.class;
    }

    @Override
    public SimpleCommitteeDTO createBasic() {
        SimpleCommitteeDTO dto = new SimpleCommitteeDTO();
        dto.setName(unique("Committee"));
        dto.setDescription("Test committee description");
        return dto;
    }

    public SimpleCommitteeDTO createWithName(String name) {
        SimpleCommitteeDTO dto = createBasic();
        dto.setName(name);
        return dto;
    }

    public SimpleCommitteeDTO createWithDetails(String name, String description) {
        SimpleCommitteeDTO dto = createBasic();
        dto.setName(name);
        dto.setDescription(description);
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/committee/SimpleCommitteeDTOFactory.kt
package net.blueshell.api.factory.dto.committee

import net.blueshell.api.committee.api.dto.SimpleCommitteeDTO
import net.blueshell.api.factory.dto.BaseDtoFactory
import org.springframework.stereotype.Component

/**
 * Factory for SimpleCommitteeDTO test instances.
 */
@Component
class SimpleCommitteeDTOFactory : BaseDtoFactory<SimpleCommitteeDTO>() {

    override fun targetType(): Class<SimpleCommitteeDTO> = SimpleCommitteeDTO::class.java

    override fun createBasic(): SimpleCommitteeDTO {
        val dto = SimpleCommitteeDTO()
        dto.name = unique("Committee")
        dto.description = "Test committee description"
        return dto
    }

    fun createWithName(name: String): SimpleCommitteeDTO {
        val dto = createBasic()
        dto.name = name
        return dto
    }

    fun createWithDetails(name: String, description: String): SimpleCommitteeDTO {
        val dto = createBasic()
        dto.name = name
        dto.description = description
        return dto
    }
}
